src/shared/components/palette/document/read-document-field.component.ts

```diff
@@ -1,8 +1,30 @@
 import { Component } from '@angular/core';
 import { AbstractFieldReadComponent } from '../base-field/abstract-field-read.component';
+import { Router } from '@angular/router';
+import { WindowService } from '../../../services/window';
+import { DocumentManagementService } from '../../../services/document-management';
+
+const MEDIA_VIEWER = 'media-viewer';
 
 @Component({
   selector: 'ccd-read-document-field',
   templateUrl: './read-document-field.html'
 })
-export class ReadDocumentFieldComponent extends AbstractFieldReadComponent {}
+export class ReadDocumentFieldComponent extends AbstractFieldReadComponent {
+
+  constructor(private windowService: WindowService,
+              private documentManagement: DocumentManagementService,
+              private router: Router) {
+    super();
+  }
+
+  showMediaViewer(): void {
+    let routerMediaViewer = this.router.createUrlTree(['/media-viewer']);
```

Comment:
shall we rename this `mediaViewerUrl`?

Reply:
or we could introduce a private method `getMediaViewerUrl(this.router)`

Reply:
`.toString()` will fail if the router returns null, so ignoring it for now

Reply:
then we should add a null check. And you are doing that in the called method anyway:
```if (urlTree) {
      window.open(urlTree.toString(), '_blank');
    }```
so better have it here then.

Reply:
Done.